docs/how-tos/validate-nexus-file.md

```diff
@@ -0,0 +1,802 @@
+
+As a python package, this can be used on Linux and Windows systems.
+
+The package can be installed via pip. Therefore you need to have installed:
+
+1. python
+
+2. pip
+
+This tool has 3 command line functions:
+
+1. [dataconverter](<https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/README.md>)
+
+2. [read\_nexus](<https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/nexus/README.md>)
+
+3. [generate\_eln](<https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/eln_mapper/README.md>)
+
+4. [verify\_nexus](https://github.com/FAIRmat-NFDI/pynxtools/pull/333)
+
+
+For validation purposes, we will use the "read\_nexus" and "verify\_nexus" function.
+
+# 1.1 verify_nexus
+
+This tool is currently in development. It enables a command like:
+
+```
+verify_nexus C:\nexusvalidation\Raman.nxs
+```
+
+
+The output warning looks like this:
+```
+...
+WARNING: Field /entry/instrument/beam_incident/wavelength/@units written without documentation.
+...
+```
+
+# 1.1.1 Installation of verify_nexus
+
+Aside from read_nexus, there is a second functionality from pynxtools. This method is currently in development (Aug 2024). This function is called: verify\_nexus. Therefore, you have to use the development install, until this function is published.
+
+Do this to install pynxtools with verify\_nexus
+
+```
+git clone https://github.com/FAIRmat-NFDI/pynxtools/tree/hdf-based-validation
+cd pynxtools
+git checkout git checkout hdf-based-validation
+git submodule sync --recursive
+git submodule update --init --recursive --jobs=4
+python -m pip install --upgrade pip
+python -m pip install -e .
+python -m pip install -e ".[dev]"
+```
+
+then, the help function should be available:
+
+```
+verify_nexus --help
+```
+
+with this output:
+
+```
+Usage: verify_nexus [OPTIONS] FILE
+
+  Verifies a nexus file
+
+Options:
+  --help  Show this message and exit.
+```
+
+
+# 1.1.2 Using verify_nexus
+
+Open your terminal. Assuming there is a folder at:
+
+For Linux:
+
+```
+/home/USER/nexusvalidation
+```
+
+For Windows:
+
+```
+C:\nexusvalidation
+```
+
+Put into this folder your NeXus file, for example this [Raman.nxs file](https://zenodo.org/records/13373909/files/Raman.nxs?download=1).
+
+Use verify_nexus with the command:
+
+```
+verify_nexus C:\nexusvalidation\Raman.nxs
+```
+
+The respective output is:
+
+```
+WARNING: Field /entry/data/spectrum_data_x/@units written without documentation.
+WARNING: Field /entry/data/spectrum_data_x_Raman/@units written without documentation.
+WARNING: Field /entry/data/spectrum_data_y/@units written without documentation.
+WARNING: Field /entry/instrument/beam_incident/wavelength/@units written without documentation.
+WARNING: Field /entry/instrument/detector_DU970BV/number_of_cycles/@units written without documentation.
+Invalid: The entry `entry` in file `Raman.nxs` is NOT a valid file according to the `NXraman` application definition.
+```
+
+
+# 1.2 read_nexus
+
```

Comment:
Explain here why there's two tools at all:
```suggestion
While `verify_nexus´ is used as a straightforward tool for validating a NeXus file, `read_nexus` outputs a debug log for a given NeXus file by annotating the data and metadata entries with the schema definitions from the respective NeXus base classes and application definitions to which the file refers to. This can be helpful to extract documentation and understand the concept defined in the NeXus application definition.
```